llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -57594,6 +57599,86 @@ static SDValue combinePDEP(SDNode *N, SelectionDAG &DAG,
   return SDValue();
 }
 
+// Fixup the MMX intrinsics' types: in IR they are expressed with <1 x i64>,
```

Comment:
No -- the problem is that DAGtoDAG is after LegalizeTypes, and v1i64 is not a legal type. LegalizeTypes can't do anything to legalize the vector type arg to an intrinsic, so we need to fix the operand/result types of the intrinsic first, before LegalizeTypes sees it.

Reply:
Can we use `i64` instead of `<1 x i64>` given we don't care about the ABI?

Reply:
We perhaps _could_ use i64, but i64 is also not a legal type on 32bit x86, so I don't think that would actually help here.

I chose <1 x i64> mainly because that's what Clang was already using for MMX values.

Reply:
Right. I forgot the 32bit case.